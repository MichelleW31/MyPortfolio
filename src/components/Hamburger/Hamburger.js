import React from "react";
import "./Hamburger.scss";

const Hamburger = ({ showMobileMenu }) => {
  const toggleMenu = () => {
    const topNavBar = document.querySelector(".topnavbar");
    const middleNavBar = document.querySelector(".middlenavbar");
    const bottomNavBar = document.querySelector(".bottomnavbar");

    topNavBar.classList.toggle("topclick");
    middleNavBar.classList.toggle("middleclick");
    bottomNavBar.classList.toggle("bottomclick");

    showMobileMenu();
  };

  return (
    <div className="hamburger_menu" onClick={toggleMenu}>
      <div className="bar topnavbar transition"></div>
      <div className="bar middlenavbar transition"></div>
      <div className="bar bottomnavbar transition"></div>
    </div>
  );
};

export default Hamburger;
