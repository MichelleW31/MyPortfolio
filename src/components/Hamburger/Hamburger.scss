@import "../../_variables.scss";

.hamburger_menu {
  display: flex;
  flex-basis: 10%;
  flex-direction: column;
  justify-content: space-between;
  position: relative;

  .bar {
    background: $black;
    border-radius: 6px;
    height: 20%;
    width: 90%;
  }

  .transition {
    transition: transform 0.5s, right 0.5s;
    transform-origin: right;
  }

  .topclick {
    transform: rotate(-39deg);
  }

  .middleclick {
    height: 0;
    right: -185px !important;
    width: 0;
  }

  .bottomclick {
    transform: rotate(39deg);
  }

  .middlenavbar {
    top: 0;
    right: 0;
    position: relative;
    transition: height 0.5s, width 0.5s, transform 0.5s, right 0.5s;
  }
}
