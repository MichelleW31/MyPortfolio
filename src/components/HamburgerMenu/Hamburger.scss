.hamburger_menu {
  display: flex;
  flex-basis: 10%;
  flex-direction: column;
  justify-content: space-between;

  .bar {
    background-color: white;
    width: 100%;
    height: 20%;
    border-radius: 6px;
  }
}
