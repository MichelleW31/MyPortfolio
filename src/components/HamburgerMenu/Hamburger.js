import React from "react";
import "./Hamburger.scss";

const Hamburger = () => {
  return (
    <div className="hamburger_menu">
      <div className="bar topbar"></div>
      <div className="bar middlebar"></div>
      <div className="bar bottombar"></div>
    </div>
  );
};

export default Hamburger;
