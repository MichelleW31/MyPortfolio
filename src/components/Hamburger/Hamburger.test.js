import React from "react";
import renderer from "react-test-renderer";
import Hamburger from "./Hamburger";

describe("Hamburger Component", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<Hamburger />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
