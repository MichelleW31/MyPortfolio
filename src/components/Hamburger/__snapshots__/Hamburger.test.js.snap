// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger Component matches the snapshot 1`] = `
<div
  className="hamburger_menu"
  onClick={[Function]}
>
  <div
    className="bar topnavbar transition"
  />
  <div
    className="bar middlenavbar transition"
  />
  <div
    className="bar bottomnavbar transition"
  />
</div>
`;
